graphql/errors.go

```diff
@@ -0,0 +1,14 @@
+package graphql
+
+import "fmt"
+
+// HTTPError represents an HTTP error with status code and response body.
+type HTTPError struct {
+	Body       string
+	StatusCode int
+}
+
+// Error implements the error interface for HTTPError.
+func (e *HTTPError) Error() string {
+	return fmt.Sprintf("returned error %v: %s", e.StatusCode, e.Body)
```

Comment:
This error-text can be hard to parse when the body is the empty string (or just whitespace), so you may want to do `%#v` instead of `%s`, or something like `>>>%s<<<`.  Though maybe it's not going to matter much in practice.

Reply:
I think it needs to be `%s` to convert the bytes to string, but I can at least add quotes!